provider/azure/internal/ad/client.go

```diff
@@ -69,8 +77,30 @@ func WithOdataErrorUnlessStatusCode(codes ...int) autorest.RespondDecorator {
 
 				// Copy and replace the Body in case it does not contain an error object.
 				// This will leave the Body available to the caller.
-				b, decodeErr := autorest.CopyAndDecode(autorest.EncodedAsJSON, resp.Body, &oe)
+				var b bytes.Buffer
+				if _, err := io.Copy(&b, resp.Body); err != nil {
+					return err
+				}
+				var decodeErr error
```

Comment:
Given that we're now reading the whole body, why not:

     var utf8BOM = []byte("\ufeff")    // global
     data, err := ioutil.ReadAll(resp.Body)
     if err != nil ...
     if err := json.Unmarshal(bytes.TrimPrefix(data, utf8BOM), &oe); err != nil ... {
     }
     resp.Body = io.NopCloser(bytes.NewReader(data))
?

BTW I think it might be better to strip the BOM in all JSON decoding code - surely
this isn't the only place that can include it?